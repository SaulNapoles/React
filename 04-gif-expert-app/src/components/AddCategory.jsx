import { useState } from 'react';


export const AddCategory = ( { setCategories } ) => {

    const [ inputValue, setInputValue ] = useState();
    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        if( inputValue.trim().length <= 0 ) return;
        setCategories( categories => [ inputValue, ...categories] );
        setInputValue('')
    }
    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Buscar GIFs"
                value =  { inputValue }
                onChange = { onInputChange }
            />
        </form>
    )
}
