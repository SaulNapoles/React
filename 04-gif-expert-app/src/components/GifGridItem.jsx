import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url, id }) => {

  return (
    <div className="card" >
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

/*
 * 1. AÃ±adir PropTypes  
 * a. title obligatorio
 * b. url obligatorio
 * 
 * 2. Evaluar el snapshot
 * 
 */

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}