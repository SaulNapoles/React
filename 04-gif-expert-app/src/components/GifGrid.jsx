// Importaciones de React primero

// Despues Importaciones de componentes o hooks
import { GifGridItem } from "./GifGridItem";
// Luego Importaciones de nuetro codigo
import { useFetchGifs } from "../hooks/useFetchGifs"; //primero los custom Hooks


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
        
        <h3>{ category }</h3>    
        {
            isLoading && (<h2>Cargando...</h2>) // And logico; si isLoaging es True, ejecuta la siguiente parte despues de &&
        }

        <div className="card-grid">
            {
                images.map(  ( image ) => (
                    <GifGridItem 
                        key={ image.id }
                        { ...image }
                    />
                ))
            }
        </div>

        </>
    )
}