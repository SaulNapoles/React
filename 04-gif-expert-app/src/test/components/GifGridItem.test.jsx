import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas <GifGridItem/>', () => { 
    const title = "Saitama";
    const url = 'http://localhost/'

    test('debe hacer match con el snapshot', () => { 

        const {container} = render(<GifGridItem title ={ title } url={ url } />);
        expect( container ).toMatchSnapshot();

    });

   test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
      
        render( <GifGridItem title ={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url);
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })


    test('debe de mostrar el titulo en el componente', () => { 
        render( <GifGridItem title ={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
     });
    
})