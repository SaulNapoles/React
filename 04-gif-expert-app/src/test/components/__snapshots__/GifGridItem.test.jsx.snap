// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://localhost/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
